3-0-spring-framework/3-3-0-enable-string-trimming/src/test/java/com/bobocode/TrimmingPostProcessorTest.java

```diff
@@ -0,0 +1,254 @@
+        assertTrue(beanAnnotation.isPresent());
+    }
+
+    @Test
+    @Order(9)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor implements BeanPostProcessor interface")
+    void trimmedAnnotationBeanPostProcessor_implementsBeanPostProcessor() {
+        boolean isBeanPostProcessorPresent = Arrays.stream(TrimmedAnnotationBeanPostProcessor.class.getInterfaces())
+                .anyMatch(i -> i.isAssignableFrom(
+                        BeanPostProcessor.class));
+        assertTrue(isBeanPostProcessorPresent);
+    }
+
+    @Test
+    @Order(10)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor overrides postProcessAfterInitialization() method")
+    void trimmedAnnotationBeanPostProcessor_overridesPostProcessAfterInitMethod() {
+        List<String> methodNames = Arrays.stream(TrimmedAnnotationBeanPostProcessor.class.getDeclaredMethods())
+                .map(Method::getName)
+                .toList();
+        assertThat(methodNames, contains("postProcessAfterInitialization"));
+    }
+
+    @Test
+    @Order(11)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor does not override postProcessBeforeInitialization() method")
+    void trimmedAnnotationBeanPostProcessor_notOverridesPostProcessAfterInitMethod() {
+        boolean isBeanPostProcessorPresent = Arrays.stream(TrimmedAnnotationBeanPostProcessor.class.getInterfaces())
+                .anyMatch(i -> i.isAssignableFrom(
+                        BeanPostProcessor.class));
+        if (!isBeanPostProcessorPresent) {
+            throw new AssertionError("TrimmedAnnotationBeanPostProcessor must implement BeanPostProcessor interface");
+        }
+
+        boolean isBeforeProcessingMethodPresent = Arrays.stream(TrimmedAnnotationBeanPostProcessor.class.getDeclaredMethods())
+                .anyMatch(m -> m.getName().equals("postProcessBeforeInitialization"));
+
+        assertFalse(isBeforeProcessingMethodPresent);
+    }
+
+    @Test
+    @Order(12)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor trims String input params marked by @Trimmed if " +
+            "trimming is enabled by @EnableStringTrimming ")
+    void trimmedAnnotationPostProcessor_annotatedInputParams(@Autowired TrimmedService service) {
+        String inputArgs = " Simba Bimba  ";
+        String actual = service.getTheTrimmedString(inputArgs);
+
+        assertEquals(inputArgs.trim(), actual);
+    }
+
+    @Test
+    @Order(13)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor not trims String input params that not marked by @Trimmed")
+    void trimmedAnnotationPostProcessor_notAnnotatedInputParams(@Autowired TrimmedService service) {
+        checkIfBeanPostProcessorImplemented();
+
+        String inputArgs = "  Simba Bimba  ";
+        String actual = service.getNotTrimmedString(inputArgs);
+
+        assertEquals(inputArgs, actual);
+    }
+
+    @Nested
+    @SpringJUnitConfig(classes = TrimmingNotEnabledConfig.class)
```

Comment:
@Stanislav-Zabramnyi as far as I understand, a nested Spring test will inherit Spring context from the parent class, so the `TrimmedAnnotationBeanPostProcessor` will be registered. 😕 Probably that's another reason to **make a parent test class simple (without spring context).**

Reply:
But how than I can test the @EnableStrinTrimming annotation?